src/op.cpp

```diff
@@ -1287,8 +1287,12 @@ uint32_t jitc_var_new_op(JitOp op, uint32_t n_dep, const uint32_t *dep) {
                     stmt = literal_zero[2] ? "selp.$b0 $r0, $r2, 0, $r1"
                                            : "selp.$t0 $r0, $r2, $r3, $r1";
                 } else {
-                    stmt = "@$r1 mov.pred $r0, $r2$n"
-                           "@!$r1 mov.pred $r0, $r3";
+                    stmt = "@!$r1 bra l_$i0_masked$n"
+                           "mov.pred $r0, $r2$n"
+                           "bra.uni l_$i0_done;\n"
+                           "l_$i0_masked:\n    "
+                           "mov.pred $r0, $r3;\n"
+                           "l_$i0_done:";
```

Comment:
@wjakob this is for the `dr.select` statement.

Reply:
That seems a bit extreme to use jumps here. Can we express this using binary logic? Basically we need PTX instructions for the ternary operator (a ? b : c). For example: `(a && b) || (!a && c)`. Maybe there is a more efficient way.

Reply:
Something like this maybe?
``` 
                    // r0 = (r1 && r2) || (!r1 && r3)
                    stmt = "and.$b0 $r2, $r1, $r2$n"
                           "not.$b0 $r1, $r1$n"
                           "and.$b0 $r3, $r1, $r3$n"
                           "or.$b0 $r0, $r2, $r3$n";
```